modules/tracktion_engine/playback/graph/tracktion_LoopingMidiNode.cpp

```diff
@@ -1063,7 +1071,19 @@ class LoopedMidiEventGenerator : public MidiGenerator
 
         loopIndex = newLoopIndex;
         const auto sequenceOffset = clipRange.getStart() + (loopIndex * loopTimes.getLength());
-        generator->cacheSequence (sequenceOffset, loopTimes + sequenceOffset);
+        generator->cacheSequence(sequenceOffset, loopTimes + sequenceOffset);
+        
+        // Get notes that would be active at the start of the new loop
+        ActiveNoteList notesAtLoopStart = generator->getNotesOnAtTime(loopTimes.getStart(),
+                                                  channelNumbers,
+                                                  clipLevel);
+
+        activeNoteList->reset();
+
+        // Track which notes we're preserving vs adding new
+        notesAtLoopStart.iterate([&](int channel, int note) {
+            activeNoteList->startNote(channel, note);
+        });
```

Comment:
So it looks like this is the most important part of the fix?
I'm not sure it's using the right variables though, `activeNoteList` is supposed to track the notes that have been started/stopped so resetting it without explicitly sending note-off events means it would get out of sync?

Are we sure that the only notes that are "active" are those `notesAtLoopStart`?
I think I'd prefer to make a change that turns off active notes then uses the `notesAtLoopStart` to turn on (and send) the correct note-ons?

The way you've done it just feels a bit brittle?